### PR TITLE
another clang-format iteration for FWCore/Framework

### PR DESCRIPTION
**Pull request by davidlange6**
Relative to the last iterations - adjusting the treatment of comments (trying to leave multiline comments alone) and template (not requiring line breaks)

I just did FWCore/Framework this time. I'd propose to discuss potential deficiencies in the next (or next to next) core meeting..

### CODE REVIEW

**Comment by davidlange6** [comment]
thats interesting..

-    assert(parsed[2].second == "");
+    assert(parsed[2].second.empty());

File: FWCore/Framework/interface/ComponentFactory.h [hunk 1]
```diff
@@ -34,100 +34,94 @@
...
-namespace edm { namespace eventsetup { \
-template<> edm::eventsetup::ComponentFactory<_type_> const* edm::eventsetup::ComponentFactory<_type_>::get() \
-{ return &s_dummyfactory; } \
-  } } \
-typedef int componentfactory_get_needs_semicolon
+    };
+
+  }  // namespace eventsetup
+}  // namespace edm
+#define COMPONENTFACTORY_GET(_type_)                                                                                  \
```
Comment (Dr15Jones) [inline FWCore/Framework/interface/ComponentFactory.h]: Does this work? It removed the line continuation!

Comment (fwyzard) [reply FWCore/Framework/interface/ComponentFactory.h]: Isn't it just moved all to way the right ?

File: FWCore/Framework/interface/Callback.h [hunk 2]
```diff
@@ -27,91 +27,81 @@
...
+        proxyData_[produce::find_index<TReturn, DataT>::value] = iData;
+      }
+
+      void storeReturnedValues(TReturn iReturn) {
+        //std::cout <<" storeReturnedValues "<< iReturn <<" " <<iReturn->value_ <<std::endl;
+        using type = typename produce::product_traits<TReturn>::type;
+        setData<typename type::head_type, typename type::tail_type>(iReturn);
+      }
+
+      template <class RemainingContainerT, class DataT, class ProductsT> void setData(ProductsT& iProducts) {
```
Comment (makortel) [inline FWCore/Framework/interface/Callback.h]: If the options for newlines in template declarations are this or https://github.com/cms-sw/cmssw/pull/25529#discussion_r243341127, I think I'd prefer #25529's approach more (i.e. I 'd like to see a newline here more than avoiding a newline for short template declarations).

File: FWCore/Framework/interface/ComponentFactory.h (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/Framework/interface/ComponentFactory.h]: I retract my statement :). Thanks Andrea!

**Pull request closed without merging**